src/sidebar/util/group-organizations.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+ // TODO: Update when this is a property available on the API response
+const DEFAULT_ORG_ID = '__default__';
+
+/**
+ * Generate consistent object keys for organizations so that they
+ * may be sorted
+ *
+ * @param {Object} organization
+ * @return {String}
+ */
+function orgKey (organization) {
+  if (organization.id === DEFAULT_ORG_ID) {  return DEFAULT_ORG_ID; }
+  return `${organization.name.toLowerCase()}${organization.id}`;
+}
+
+/**
+ * Add a clone of the group object to the given organization object's
+ * groups Array.
+ *
+ * @param {Group} group
+ * @param {Object} organization
+ * @return undefined - organization is mutated in place
+ */
+function addGroup (group, organization) {
+  // Object.assign won't suffice because of nested objects on groups
+  const groupObj = JSON.parse(JSON.stringify(group));
```

Comment:
What's the problem with doing a shallow clone of `group` here? You are not modifying any of the nested fields as far as I can see.

Reply:
I don't want to be handing back references in general—I don't "have control" over what the consumer is going to do with the resulting data and would prefer to give them a real clone versus a hybrid. If your opinion is strong here, I'll switch it to a shallow clone.

Reply:
Update: I saw your later comment about `seamless-immutable` so I am OK with shallow cloning now.

Reply:
You can freeze the result with `immutable(...)` if you like and just specify that the consumer can't change it. This will save a few cycles when computing this list. If there is a ever a use case where that is particularly inconvenient for some reason we can always revisit.